sbi/inference/posteriors/base_posterior.py

```diff
@@ -173,7 +179,49 @@ def map(
             save_best_every=save_best_every,
             show_progress_bars=show_progress_bars,
         )[0]
-        return self.map_
+
+    @abstractmethod
+    def map(
+        self,
+        x: Optional[Tensor] = None,
+        num_iter: int = 1_000,
+        num_to_optimize: int = 100,
+        learning_rate: float = 0.01,
+        init_method: Union[str, Tensor] = "posterior",
+        num_init_samples: int = 1_000,
+        save_best_every: int = 10,
+        show_progress_bars: bool = False,
+        force_update: bool = False,
+    ) -> Tensor:
+        """Returns stored maximum-a-posterior estimate (MAP). Calculates MAP if none is
+        stored.
+
+        See child classes for docstring.
+        """
+
+        if x is not None:
+            raise ValueError(
+                "Passing `x` directly to `.map()` has been deprecated."
+                "Use `.self_default_x()` to set `x`, and then run `.map()` "
+            )
+
+        elif self.default_x is None:
```

Comment:
I think it's confusing that these are `elif`. Maybe just use `if`? Same in the next if case